### PR TITLE
Zoom refactor

### PR DESCRIPTION
**Pull request by IntelOrca**
This is #10807 but without actually updating the maximum zoom level. The refactor is large and widespread and likely to need rebasing a lot otherwise.

### CODE REVIEW

File: src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp [hunk 1]
```diff
@@ -686,10 +688,10 @@ void OpenGLDrawingContext::DrawSprite(uint32_t image, int32_t x, int32_t y, uint
     right -= _dpi->x;
     bottom -= _dpi->y;
 
-    left /= zoomLevel;
-    top /= zoomLevel;
-    right /= zoomLevel;
-    bottom /= zoomLevel;
+    left = left / _dpi->zoom_level;
```
Comment (Gymnasiast) [inline src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp]: Why this change?

Comment (IntelOrca) [reply src/openrct2-ui/drawing/engines/opengl/OpenGLDrawingEngine.cpp]: You can't add new `/=` operators for types like `int`.

**Comment by AaronVanGeffen** [comment]
I'm getting some compiler errors when compiling this with GCC 9:
```
/usr/bin/c++ --version
c++ (Arch Linux 9.3.0-1) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
```
```
[5/288] Building CXX object CMakeFiles/libopenrct2.dir/src/openrct2/interface/Viewport.cpp.o
FAILED: CMakeFiles/libopenrct2.dir/src/openrct2/interface/Viewport.cpp.o 
/usr/bin/c++  -DOPENGL_NO_LINK -DUSE_BENCHMARK -D__ENABLE_LIGHTFX__ -D__WARN_SUGGEST_FINAL_METHODS__ -D__WARN_SUGGEST_FINAL_TYPES__ -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -Wno-clobbered -fstrict-overflow -fstrict-aliasing -Werror -Wundef -Wmissing-declarations -Winit-self -Wall -Wextra -Wshadow -Wno-unknown-pragmas -Wno-missing-braces -Wno-comment -Wnonnull -Wno-unused-parameter -DDEBUG=0 -fPIC   -std=gnu++17 -MD -MT CMakeFiles/libopenrct2.dir/src/openrct2/interface/Viewport.cpp.o -MF CMakeFiles/libopenrct2.dir/src/openrct2/interface/Viewport.cpp.o.d -o CMakeFiles/libopenrct2.dir/src/openrct2/interface/Viewport.cpp.o -c ../src/openrct2/interface/Viewport.cpp
../src/openrct2/interface/Viewport.cpp: In function ‘void viewport_redraw_after_shift(rct_drawpixelinfo*, rct_window*, rct_viewport*, int32_t, int32_t)’:
../src/openrct2/interface/Viewport.cpp:251:63: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  251 |         std::memcpy(&view_copy, viewport, sizeof(rct_viewport));
      |                                                               ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
../src/openrct2/interface/Viewport.cpp:303:63: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  303 |         std::memcpy(viewport, &view_copy, sizeof(rct_viewport));
      |                                                               ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
../src/openrct2/interface/Viewport.cpp: In function ‘void viewport_move(int16_t, int16_t, rct_window*, rct_viewport*)’:
../src/openrct2/interface/Viewport.cpp:439:59: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  439 |     std::memcpy(&view_copy, viewport, sizeof(rct_viewport));
      |                                                           ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
../src/openrct2/interface/Viewport.cpp:458:63: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  458 |         std::memcpy(viewport, &view_copy, sizeof(rct_viewport));
      |                                                               ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
../src/openrct2/interface/Viewport.cpp:479:63: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  479 |         std::memcpy(viewport, &view_copy, sizeof(rct_viewport));
      |                                                               ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
../src/openrct2/interface/Viewport.cpp:489:59: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct rct_viewport’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
  489 |     std::memcpy(viewport, &view_copy, sizeof(rct_viewport));
      |                                                           ^
In file included from ../src/openrct2/interface/Viewport.h:14,
                 from ../src/openrct2/interface/Viewport.cpp:10:
../src/openrct2/interface/Window.h:92:8: note: ‘struct rct_viewport’ declared here
   92 | struct rct_viewport
      |        ^~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Comment by AaronVanGeffen** [comment]
Tested it again. After the GCC9 patches, all seems to work fine for me on GCC9 as well :)
